src/toil/cwl/utils.py

```diff
@@ -219,3 +221,80 @@ def download_structure(
             # TODO: why?
             index[dest_path] = value
             existing[value] = dest_path
+
+
+def trim_mounts_op_down(file_or_directory: CWLObjectType) -> None:
+    """
+    No-op function for mount-point trimming.
+    """
+    return
+
+
+def sniff_location(file_or_directory: CWLObjectType) -> Optional[str]:
+    """
+    Get the local bare path or file: URI for a CWL file or directory, or None.
+
+    :return: None if we don't have a local path or file URI
+    """
+    # Since we only consider mountable paths, if path is not file URI or bare path, don't consider it
+    path = unquote(cast(str, file_or_directory.get('location') or file_or_directory.get('path')))
```

Comment:
We can't `unquote` the path when it's not a URL, or we won't be able to work on files actually named e.g. `%25.txt`.

We also might want to do the unquote *after* parsing the URL. It really only would matter for cases where e.g. a slash character appears inside a query string value, which can't really apply to files, but I think it's technically correct.